packages/gatsby-plugin-sitemap/src/gatsby-node.js

```diff
@@ -83,7 +83,7 @@ exports.onPostBuild = async (
   }
 
   const sitemapWritePath = path.join(`public`, output)
-  const sitemapPublicPath = path.posix.join(pathPrefix, output)
+  const sitemapPublicPath = path.posix.join(basePath, output)
```

Comment:
I think this needs to be fixed but not in this way.

When using path and Asset prefixes this is rendering: `https://mysite.com/cdn.example.com/foo/sitemap/sitemap-0.xml`

This is wrong but I do believe it should still have the asset prefix and read: `https://cdn.example.com/foo/sitemap/sitemap-0.xml`

According to https://www.gatsbyjs.com/docs/how-to/previews-deploys-hosting/asset-prefix/ everything that's not HTML should be available on the assetPrefix path. That would include sitemaps. 

There is DEFINITELY an issue as the URL is mangled, but I believe we need to keep the `assetPrefix`.

Do you agree?

Reply:
I just realized the thing that's probably mangaling this is the sitemap writer. It's expecting a path and getting an FQDN. short of fixing this upstream or re writing that implementation this may not be easy to fix. One option would be just to not use asset prefix anywhere. Meaning you current fix on this is fine but we'd need to mirror that change in `gatsby-serve.js`. I'd be interested if the core team has any thoughts on this.

Reply:
I would like to second both @GriffinJohnston and @ravindra-euphorika. I am experiencing the same issue and it's causing a slew of problems with our site and our optimization goals.